sdk/communication/communication-call-automation/test/callRecordingClient.spec.ts

```diff
@@ -166,3 +185,105 @@ describe("CallRecording Unit Tests", async function () {
+      } catch (e) {
+        console.log("Call is terminated");
+      }
+    }
+    serviceBusReceivers.forEach((receiver) => {
+      receiver.close();
+    });
+    events.forEach((callConnectionEvents) => {
+      callConnectionEvents.clear();
+    });
+    events.clear();
+    serviceBusReceivers.clear();
+    incomingCallContexts.clear();
+    await recorder.stop();
+  });
+
+  it("Creates a call, start recording, and hangs up", async function () {
+    testName = this.test?.fullTitle()
+      ? this.test?.fullTitle().replace(/ /g, "_")
+      : "create_call_start_recording_and_hang_up";
+    await loadPersistedEvents(testName);
+
+    const callInvite: CallInvite = { targetParticipant: testUser2 };
+    const uniqueId = await serviceBusWithNewCall(testUser, testUser2);
+    const callBackUrl: string = dispatcherCallback + `?q=${uniqueId}`;
+
+    const result = await callerCallAutomationClient.createCall(callInvite, callBackUrl);
+    const incomingCallContext = await waitForIncomingCallContext(uniqueId, 8000);
+    const callConnectionId: string = result.callConnectionProperties.callConnectionId
+      ? result.callConnectionProperties.callConnectionId
+      : "";
+    assert.isDefined(incomingCallContext);
+
+    if (incomingCallContext) {
+      await receiverCallAutomationClient.answerCall(incomingCallContext, callBackUrl);
+    }
+    const callConnectedEvent = await waitForEvent("CallConnected", callConnectionId, 8000);
+
+    assert.isDefined(callConnectedEvent);
+    callConnection = result.callConnection;
+
+    const playSource: FileSource = {
+      url: fileSourceUrl,
+      kind: "fileSource",
+    };
+
+    // Call recording can fail when no audio is in call, we will play audio to avoid that.
+    await callConnection.getCallMedia().playToAll(playSource);
+
+    const recOptions: StartRecordingOptions = {
+      recordingStateCallbackEndpointUrl: callBackUrl,
+      callLocator: {
+        id: (await callConnection.getCallConnectionProperties()).serverCallId || "",
+        kind: "serverCallLocator",
+      },
+      recordingChannel: "unmixed",
+      recordingFormat: "wav",
+      recordingContent: "audio",
+    };
+
+    const recordingStateChanged = waitForEvent("RecordingStateChanged", callConnectionId, 8000);
+    const recordingStateResult = callerCallAutomationClient.getCallRecording().start(recOptions);
+    await recordingStateChanged;
+    assert.isDefined(recordingStateChanged);
+
+    // Delay for 6 seconds, this is to let the recording state change to active
+    await new Promise((resolve) => setTimeout(resolve, 6000));
+    const recStatus = await callerCallAutomationClient
+      .getCallRecording()
+      .getState((await recordingStateResult).recordingId);
+    assert.equal(recStatus.recordingState, "active");
+    await callerCallAutomationClient
+      .getCallRecording()
+      .stop((await recordingStateResult).recordingId);
+    await callConnection.hangUp(true);
```

Comment:
NIT: You don't have to hangUp explicitly if you are not intend to test on the hangUp function, because the afterEach() method will do it for you.

Reply:
removed hangup